src/main/ts/index.ts

```diff
@@ -49,43 +50,60 @@ export class ConsulDiscoveryService implements IConsulDiscoveryService {
     })
   }
 
-  public ready (serviceName: string): IPromise<IServiceEntry> {
-    const service = this.getService(serviceName)
+  public getValueByKey (key: string) {
+    return this.ready(key, 'kv').then(({ data }) => data)
+  }
+
+  public ready (
+    serviceName: string,
+    type: IServiceType
+  ): IPromise<IServiceDiscoveryEntry> {
+    const service = this.getService(serviceName, type)
 
     if (service.promise) {
       return service.promise
     }
 
-    const {
-      resolve,
-      reject,
-      promise
-    } = promiseFactory()
+    const { resolve, reject, promise } = promiseFactory()
 
     service.promise = promise
 
     log.debug(`watcher initialized, service=${serviceName}`)
-    ConsulDiscoveryService.watchOnChange(service, resolve, reject, this.services)
+    ConsulDiscoveryService.watchOnChange(
+      service,
+      resolve,
+      reject,
+      this.services
+    )
     ConsulDiscoveryService.watchOnError(service, reject, this.services)
 
     return promise
   }
 
-  public getService (serviceName: IServiceName): IServiceEntry {
-    return this.services[serviceName] || this.createService(serviceName)
+  public getService (
+    serviceName: IServiceName,
+    type: IServiceType
+  ): IServiceEntry {
+    return this.services[serviceName] || this.createService(serviceName, type)
   }
 
-  public createService (serviceName: IServiceName): IServiceEntry {
-    const watcher = this.getWatcher(serviceName)
-    const connections = []
+  public createService (
+    serviceName: IServiceName,
+    type: IServiceType
+  ): IServiceEntry {
+    const watcher = this.getWatcher(serviceName, type)
     const service = {
+      type,
       name: serviceName,
       watcher,
-      connections,
+      data: type === 'kv' ? {} : [],
       sequentialErrorCount: 0
     }
+
+    // @ts-ignore
     this.services[serviceName] = service
```

Comment:
There's a need for composite key (type + sep + name) or separate scopes (this.store.kv[name], this.store.discovery[name])